ar5523: add basic support for backporting

bail out on kernels already shipping ar5523 (>= v3.8)

Changes-licensed-under: ISC
Signed-off-by: Stefan Lippers-Hollmann <s.l-h@gmx.de>

--- a/drivers/net/wireless/ath/ar5523/ar5523.c
+++ b/drivers/net/wireless/ath/ar5523/ar5523.c
@@ -24,6 +24,14 @@
  * out how the hardware works.  It contains a lot magic numbers because of
  * that and only has minimal functionality.
  */
+
+#include <linux/version.h>
+#if LINUX_VERSION_CODE > KERNEL_VERSION(3, 7, 999)
+#error "****************************************************************************"
+#error "This driver has been merged mainline with v3.8, don't build this out of tree"
+#error "****************************************************************************"
+#endif
+
 #include <linux/compiler.h>
 #include <linux/init.h>
 #include <linux/kernel.h>
